src/main/java/uk/gov/justice/digital/delius/service/CourtReportService.java

```diff
@@ -37,12 +38,21 @@ public CourtReportService(CourtReportRepository courtReportRepository) {
 
     public Optional<CourtReport> courtReportFor(Long offenderId, Long courtReportId) {
 
-        Optional<uk.gov.justice.digital.delius.jpa.standard.entity.CourtReport> maybeCourtReport = courtReportRepository.findByOffenderIdAndCourtReportId(offenderId, courtReportId);
-        return maybeCourtReport
-                .filter(this::notDeleted)
+        return notDeletedCourtReportFor(offenderId, courtReportId)
                 .map(CourtReportTransformer::courtReportOf);
     }
 
+    public Optional<CourtReportMinimal> courtReportMinimalFor(Long offenderId, Long courtReportId) {
+
+        return notDeletedCourtReportFor(offenderId, courtReportId)
+            .map(CourtReportTransformer::courtReportMinimalOf);
+    }
+
+    private Optional<uk.gov.justice.digital.delius.jpa.standard.entity.CourtReport> notDeletedCourtReportFor(Long offenderId, Long courtReportId) {
+        Optional<uk.gov.justice.digital.delius.jpa.standard.entity.CourtReport> maybeCourtReport = courtReportRepository.findByOffenderIdAndCourtReportId(offenderId, courtReportId);
+        return maybeCourtReport.filter(this::notDeleted);
+    }
```

Comment:
Observation, slightly nicer to move the deleted filter to repo, we have noticed a few occasions where bad/corrupt data in Delius is causing JPA errors, even for records we end up filtering, e.g where lazy loading isn't happening.

I think I have concluded that might be better to push filtering down to repo (define the soft_deleted as a boolean makes it even easier). I think there are some examples around

Reply:
OK - have seen     `Optional<Offender> findByCrnAndSoftDeletedFalse(String crn);` . Previously, I have also seen it done with a "Where" annotation on the entity itself. There's only one place something similar happens.